src/config/CloudConfig.ts

```diff
@@ -0,0 +1,37 @@
+import ICloudConfig from "../interfaces/Config/ICloudConfig";
+import {ALLOWED_CLOUD_CONFIG, INVALID_TYPE_MESSAGE} from "../constants";
+import Config from "./BaseConfig";
+
+
+
+class CloudConfig extends Config implements ICloudConfig {
+  cloudName?: string;
+  apiKey?: string;
+  apiSecret?: string;
+
+  [other:string]: unknown;
+
+  /**
+   * @param {ICloudConfig} cloudConfig
+   */
+  constructor(cloudConfig: ICloudConfig | unknown) {
+    super();
+    if (this.isCloudConfigTypeGuard(cloudConfig)) {
+      Object.assign(this, cloudConfig);
+    } else {
+      console.error(INVALID_TYPE_MESSAGE);
+    }
+  }
```

Comment:
Just to make sure I understand the logics - if I have a typo in one of the keys, the entire thing gets ignored? 
Why not ignoring just the invalid keys? 

I feel like we're doing a lot of work the TS supposed to do... which brings me to the question of "why unknown"? if we drop that, TS will handle all validations for us, no?